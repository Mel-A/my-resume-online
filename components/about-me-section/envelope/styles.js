import styled from 'styled-components'
import MediaQueries from '../../media-queries'

export const Section = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  height: 500px;
  align-items: center;
`
export const Information = styled.div`
`
export const Title = styled.h2`
`

export const Container = styled.div`
  position: relative;
  width: 550px;
  height: 166px;
  display: flex;
  justify-content: center;
  background-color: ${({ theme }) => theme.colors.malibu};
  border-radius: 0 0 5px 5px;
  box-shadow: 0 0 1px  ${({ theme }) => theme.colors.malibu} inset;
  z-index: 1;
  &::after{
    content: '';
    position: absolute;
    width: 0;
    height: 0;
    border: 0 solid transparent;
    border-width: 83px 275px;
    border-bottom-color:  ${({ theme }) => theme.colors.malibu};
    top: -100%;
    left: 0px;
  }
`
export const Paper = styled.div`
  background:  ${({ theme }) => theme.colors.white};
  position: relative;
  width: 90%;
  height: 100%;
  z-index: 1;
  border-radius: 5px;
`
export const Shape = styled.div`
  position: absolute;
  width: 0;
  height: 0;
  border: 0 solid transparent;
  border-width: 83px 275px;
  border-right-color: ${({ theme }) => theme.colors.sail};
  border-left-color:#94c8ff;
  border-bottom-color: #94c8ff;
  top: 0;
  border-radius: 0 0 5px 5px;
  transform: rotate(360deg);
  z-index: 2;
  box-shadow: 0 1px 1px rgba(0,0,0,0.25);
`