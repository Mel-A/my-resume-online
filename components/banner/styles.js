import styled from 'styled-components'

export const Container = styled.div`
  width: 100%;
`
export const Image = styled.img`
  width: 100%;
  height: 300px;
  object-fit: cover;
  object-position: top;
`