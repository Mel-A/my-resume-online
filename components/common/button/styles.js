import styled from 'styled-components'

export const StyledButton = styled.button`
  background-color: ${({ theme }) => theme.colors.azure};
  color: ${({ theme }) => theme.colors.white};
  border: none;
  border-radius: 3px;
  padding: 10px 20px;
  cursor: pointer;
  transition: all 0.2s;
  outline: none;

  &:hover{
    color: ${({ theme }) => theme.colors.azure};
    border: 1px solid ${({ theme }) => theme.colors.azure};
    background-color: inherit;
  }
`