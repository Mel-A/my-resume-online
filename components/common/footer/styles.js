import styled from 'styled-components'
import MediaQueries from '../../media-queries'

export const Container = styled.div`
  background-color: ${({ theme }) => theme.colors.black};
  height: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
`
export const LinkImage = styled.img`
  filter: invert(96%) sepia(100%) saturate(0%) hue-rotate(33deg) brightness(105%) contrast(103%);
  height: 20px;
`
export const Item = styled.label`
  color: ${({ theme }) => theme.colors.white};
  cursor: pointer;
  ${MediaQueries.mobile`
    display: none;
  `};
`
export const Link = styled.a`
  cursor: pointer;
  position: relative;
  align-items: center;
  display: flex;
  flex-direction: column;
`