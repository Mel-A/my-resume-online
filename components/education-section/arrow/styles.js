import styled, { css } from 'styled-components'
import MediaQueries from '../../media-queries'

export const Image = styled.img`
  height: 60px;
  width: 60px;
  ${({ right }) => !right && css`transform: rotate(180deg);`}

  ${MediaQueries.mobile`
    height: 30px;
    width: 30px;
  `};
`
export const Swipper = styled.div`
  position: absolute;
  cursor: pointer;
  display: flex;
  align-items: center;
  z-index: 2;
  top: calc(50% - 10px);

  ${({ right }) => right ? 
    css` 
      flex-direction: row-reverse;
      right: -12px;` : 
    css` 
      left: -12px;`
  };

  ${MediaQueries.mobile`
    top: unset;
    bottom: -10px;

    ${({ right }) => right ? 
      css` 
        right: 15%;` : 
      css` 
        left: 15%;`
    };
  `};
`
export const Indicator = styled.label`
  display: none;
  margin: 0 20px;
  ${MediaQueries.mobile`
    display: inline;
  `};
`