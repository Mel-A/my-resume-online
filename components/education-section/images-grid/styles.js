import styled from 'styled-components'
import MediaQueries from '../../media-queries'

export const Title = styled.h2`
  font-size: ${({ theme }) => theme.fonts.medium4};

  ${MediaQueries.mobile`
    font-size: ${({ theme }) => theme.fonts.medium2};
  `};
`
export const Selected = styled.div`
  justify-content: center;
  position: absolute;
  display: flex;
  flex-direction: row;
  height: ${({ visible }) => visible ? '100%' : '0%'};
  opacity: ${({ visible }) => visible ? '1' : '0'};
  transition: height 0s, opacity 1s;

  ${MediaQueries.mobile`
    height: 100%;
    flex-direction: column;
    justify-content: space-between;
  `};
`
export const Paragraph = styled.p`
  max-width: 90%;
  white-space: pre-wrap;

  ${MediaQueries.mobile`
    display: inline;
    font-size: 14px;
  `};
`
export const Carousel = styled.div`
  position: relative;
  width: 100%;
  height: 350px;
  
  ${MediaQueries.mobile`
    height: 800px;
  `};
`
export const Circle = styled.div`
  display: flex;
  border-radius: 50%;
  width: 420px;
  z-index: 0;
  height: 420px;
  justify-content: center;
  align-items: center;
  border: 5px solid ${({ theme }) => theme.colors.black};
  border-width: 1px 0 1px 10px;

  ${MediaQueries.mobile`
    max-width: 340px;
    width: 90vw;
    z-index: 0;
    height: 340px;
  `};
`
export const LinearCircle = styled.div`
  display: flex;
  border-radius: 50%;
  width: 390px;
  z-index: 0;
  height: 390px;
  border: 5px solid ${({ theme }) => theme.colors.aquamarine};
  justify-content: center;
  align-items: center;
  border-width: 1px 10px 1px 0;

  ${MediaQueries.mobile`
    max-width: 320px;
    width: 90vw;
    z-index: 0;
    height: 320px;
  `};
`
export const Image = styled.img`
  width: 350px;
  height: 350px;
  border-radius: 50%;
  z-index: 1;
  position: relative;

  ${MediaQueries.mobile`
    max-width: 300px;
    width: 80vw;
    z-index: 0;
    height: 300px;
  `};
`
export const Container = styled.div`
  width: 100%;
  max-width: 1300px;
  display: flex;
  flex-direction: row;
  position: relative;
  padding: 0 60px;
  box-sizing: border-box;
  justify-content: space-between;

  ${MediaQueries.mobile`
    justify-content: center;
    padding: 0 10px;
  `};
`
export const Column = styled.div`
  max-width: 50%;
  position: relative;

  ${MediaQueries.mobile`
    text-align: center;
    max-width: 100%;
    margin-bottom: 40px;
    display: flex;
    flex-direction: column;
    align-items: center;
  `};
`
export const Wrapper = styled.div`
  margin-top: 100px;
  background-color: ${({ theme }) => theme.colors.mercury};
  padding: 150px 0 150px;
  display: flex;
  justify-content: center;
  
  ${MediaQueries.mobile`
    overflow: hidden;
  `};
`