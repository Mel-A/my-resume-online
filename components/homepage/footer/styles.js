import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${({ theme }) => theme.colors.black};
  height: 100px;
  display: flex;
  justify-content: space-around;
  align-items: center;
`
export const LinkImage = styled.img`
  filter: invert(96%) sepia(100%) saturate(0%) hue-rotate(33deg) brightness(105%) contrast(103%);
  height: 20px;
`
export const Item = styled.label`
  color: ${({ theme }) => theme.colors.white};
`
export const Link = styled.a`
  cursor: pointer;
  position: relative;
  align-items: center;
  display: flex;
  flex-direction: column;
`