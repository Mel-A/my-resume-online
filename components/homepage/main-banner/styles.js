import styled from 'styled-components'

export const Image = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  object-fit: cover;
  z-index: 0;
`
export const Container = styled.div`
  width: 100%;
  height: 400px;
  position: relative;
  flex-direction: column;
  display: flex;
  align-items: center;
  justify-content: center;
`
export const Title = styled.h1`
  z-index: 1;
  color: ${({ theme }) => theme.colors.white};
`
export const Subtitle = styled.p`
  z-index: 1;
  color: ${({ theme }) => theme.colors.white};
`