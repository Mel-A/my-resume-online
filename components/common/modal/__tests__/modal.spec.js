import { mount, shallow } from 'enzyme'
import Modal from '..'
import { ThemeProvider } from "styled-components"
import theme from '../../../../theme'
import { Button } from '../..'
import { Icon } from '../styles'

const dummyData = {
  title: 'some title',
  onClick: jest.fn(),
  body: 'some body text'
}

jest.mock('react-i18next', () => ({
  useTranslation: () => ({t: key => key})
}))


describe('Modal', () => {

  it('modal without modalState', () => {
    const modal = shallow(<Modal {...dummyData} />)
    expect(modal.instance()).toBeNull()

  })

  it('modal renders theme correctly', () => {
    const modal = mount(<Modal {...dummyData} modalState />, {
      wrappingComponent: ThemeProvider,
      wrappingComponentProps: {
        theme,
      },
    })
    expect(modal).toMatchSnapshot()

  })

  it('onClick method for button', () => {
    const modal = mount(<Modal {...dummyData} modalState />, {
      wrappingComponent: ThemeProvider,
      wrappingComponentProps: {
        theme,
      },
    })
    modal.find(Button).simulate('click');

    expect(dummyData.onClick).toHaveBeenCalled()
  })

  it('onClick method for icon', () => {
    const modal = mount(<Modal {...dummyData} modalState />, {
      wrappingComponent: ThemeProvider,
      wrappingComponentProps: {
        theme,
      },
    })
    modal.find(Icon).simulate('click');

    expect(dummyData.onClick).toHaveBeenCalled()
  })
})