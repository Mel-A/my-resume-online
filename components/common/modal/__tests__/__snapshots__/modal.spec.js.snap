// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`banner renders correctly 1`] = `ShallowWrapper {}`;
